Shared/Samples/Snap geometry edits with utility network rules/SnapGeometryEditsWithUtilityNetworkRulesView.swift

```diff
@@ -0,0 +1,280 @@
+        
+        var body: some View {
+            NavigationStack {
+                Form {
+                    Section {
+                        ForEach(Array(settings.enumerated()), id: \.offset) { _, settings in
+                            SnapSourceSettingsToggle(
+                                settings: settings,
+                                image: ruleBehaviorImages[settings.ruleBehavior]
+                            )
+                        }
+                    }
+                    
+                    Section("Legend") {
+                        ForEach(SnapRuleBehavior?.allCases, id: \.self) { behavior in
+                            Label {
+                                Text(behavior.label)
+                            } icon: {
+                                ruleBehaviorImages[behavior]
+                            }
+                            .task(id: displayScale) {
+                                // Creates an image from the rule behavior's symbol.
+                                let swatch = try? await behavior.symbol.makeSwatch(scale: displayScale)
+                                ruleBehaviorImages[behavior] = Image(uiImage: swatch ?? UIImage())
+                            }
+                        }
+                    }
+                }
+                .navigationTitle("Snap Sources")
+                .navigationBarTitleDisplayMode(.inline)
+                .toolbar {
+                    ToolbarItem(placement: .confirmationAction) {
+                        Button("Done") { dismiss() }
+                    }
+                }
+            }
+        }
+    }
+    
+    /// A toggle for enabling and disabling a given snap source settings.
+    struct SnapSourceSettingsToggle: View {
+        /// The snap source settings to enable and disable.
+        let settings: SnapSourceSettings
+        
+        /// The image to use in the toggle's label.
+        let image: Image?
+        
+        /// A Boolean value indicating whether the toggle is enabled.
+        @State private var isEnabled = false
+        
+        var body: some View {
+            Toggle(isOn: $isEnabled) {
+                Label {
+                    Text(settings.source.name)
+                } icon: {
+                    image
+                }
+            }
+            .onChange(of: isEnabled) { newValue in
+                settings.isEnabled = newValue
+            }
+            .onAppear {
+                isEnabled = settings.isEnabled
+            }
+        }
+    }
+}
+
+// MARK: - Extensions
+
+extension SnapSource {
+    /// The name of the snap source.
+    var name: String {
+        switch self {
+        case let graphicsOverlay as GraphicsOverlay:
+            graphicsOverlay.id
+        case let layerContent as LayerContent:
+            layerContent.name
+        default:
+            "\(self)"
+        }
+    }
+}
+
+extension SnapRuleBehavior? {
```

Comment:
This was really surprising at first. Consider not using syntactic sugar here to improve readability:

```suggestion
extension Optional<SnapRuleBehavior> {
```